net/nettest/pipe.go

```diff
@@ -145,117 +148,94 @@ func (p *Pipe) Close() error {
 		p.cancelReadTimer()
 		p.cancelReadTimer = nil
 	}
-	p.mu.Unlock()
+	p.cnd.L.Unlock()
 
 	if closed {
-		return fmt.Errorf("nettest.Pipe(%q).Close: already closed", p.name)
+		return net.ErrClosed
 	}
+	p.cnd.Broadcast()
 
-	p.signalRead()
-	p.signalWrite()
 	return nil
 }
 
+func (p *Pipe) deadlineTimer(t time.Time) func() {
+	if t.IsZero() {
+		return nil
```

Comment:
ideally you'd also track the outstanding goroutine from before and shut it down.

likewise when the pipe is closed.

Reply:
oh! you do!

I see now.